upgrade_test.py

```diff
@@ -828,6 +831,133 @@ def test_rolling_upgrade(self):  # pylint: disable=too-many-locals,too-many-stat
 
         InfoEvent(message='all nodes were upgraded, and last workaround is verified.').publish()
 
+    def test_custom_profile_rolling_upgrade(self):  # pylint: disable=too-many-locals,too-many-statements
+        """
+        Run a load of a custom profile.
+        Upgrade half of nodes in the cluster, and run rollback.
+        Upgrade all nodes to new version in the end.
+        """
+        InfoEvent(message='Running a prepare load for the initial custom data').publish()
+        if not (prepare_cs_user_profiles := self.params.get('prepare_cs_user_profiles')):
+            SyntaxError("Parameter 'prepare_cs_user_profiles' is not supplied")
+
+        user_profiles, durations = self.parse_cs_user_profiles_param(prepare_cs_user_profiles)
+        stress_before_upgrade, _ = self.run_cs_user_profiles(cs_profiles=user_profiles, duration=durations)
+        for queue in stress_before_upgrade:
+            self.verify_stress_thread(cs_thread_pool=queue)
+
+        # write workload during entire test
+        InfoEvent(message='Starting write workload during entire test').publish()
+        if not (cs_user_profiles := self.params.get('cs_user_profiles')):
+            SyntaxError("Parameter 'cs_user_profiles' is not supplied")
+
+        user_profiles, durations = self.parse_cs_user_profiles_param(cs_user_profiles)
+        entire_write_thread_pool, stress_cmds = self.run_cs_user_profiles(cs_profiles=user_profiles, duration=durations)
+
+        if "scylla-qa-internal" in ",".join(cs_user_profiles):
+            stress_cmds = cs_user_profiles
+        self.log.debug("stress_cmds: %s", stress_cmds)
+
+        # Let to write_stress_during_entire_test complete the schema changes
+        for user_profile in stress_cmds:
+            _, profile = get_profile_content(user_profile)
+            keyspace_name = profile.get('keyspace')
+            if not keyspace_name:
+                continue
+
+            try:
+                self.metric_has_data(
```

Comment:
we are assuming each stress might have different keyspace, they why are we checking this inside the loop ?

Reply:
Yes, I did it for this case